### ISSUE TITLE
Incorrect behaviour when running multiple batch/cmd files in a single command step

### ISSUE THREAD

**Issue by aknarayan** [buildkite/agent#1538]
Agent version 3.33.3

We had a scenario where we needed to run two cmd files, `Setup.cmd` and `BuildProject.cmd`, in a single command step on Windows. The BK agent writes these into a single batch file as follows:
![image](https://user-images.githubusercontent.com/46627784/143615822-395219ed-2219-486f-b293-ddfd02b09316.png)
However, this generated batch file is incorrect. When calling batch/cmd files from other batch/cmd files, the `call` keyword should be used, otherwise multiple files are not called correctly and only the first file is called. In the above example `Setup.cmd` is run and then the command completes; `BuildProject.cmd` is never run. The correct batch file should be:
```
@echo off
call Setup.cmd
if %errorlevel% neq 0 exit /b %errorlevel%
call BuildProject.cmd
if %errorlevel% neq 0 exit /b %errorlevel%
```
The workaround here is to wrap the calls to multiple files in a single batch/cmd file and only run that single file in the command step.

(cc @PaulSonOfLars @elizabethking2)

**Comment by jkoooo4456** [comment]
I

**Comment by keithduncan** [comment]
Thank you for this report @aknarayan 😄 we’ve merged the fix for this in #1543 and it will go out in the next agent release :bow:
